static/src/javascripts/projects/commercial/modules/carrot-slot.js

```diff
@@ -4,22 +4,43 @@ import { addSlot } from 'commercial/modules/dfp/add-slot';
 import createSlot from 'commercial/modules/dfp/create-slot';
 import { commercialFeatures } from 'commercial/modules/commercial-features';
 
+/*
+ The heuristic for adding a carrot slot into an article is as follows:
+ 1) it must be after the second paragraph and
+ 2) it must be after the 100th word in the article
+
+ This is to prevent it appearing after single word paragraphs that are encountered
+ at the top of some article. It still isn't perfect, but it is good enough for a test.
+ */
+const findSuitableCarrotPosition = (
+    paragraphs: Array<HTMLElement>
+): ?HTMLElement => {
+    let runningWordCount = 0;
+    let index = 0;
+
+    while (index < paragraphs.length) {
+        runningWordCount += paragraphs[index].textContent.split(' ').length;
+
+        if (runningWordCount >= 100 && index >= 1) return paragraphs[index];
+
+        index += 1;
+    }
+};
+
 const carrotSlotInit = () => {
     if (commercialFeatures.carrotSlot) {
-        const anchorSelector = '.js-carrot';
-        const anchor = document.querySelector(anchorSelector);
-
-        const slot = createSlot('carrot');
-
-        return fastdom
-            .write(() => {
-                if (anchor) {
-                    anchor.insertAdjacentElement('beforebegin', slot);
-                }
-            })
-            .then(() => {
-                addSlot(slot, true);
-            });
+        const paragraphs = Array.from(
+            document.querySelectorAll('.js-article__body p')
+        );
+        const anchor: ?HTMLElement = findSuitableCarrotPosition(paragraphs);
```

Comment:
I've rewritten this logic to now use the space-finder - would be great to get your thoughts on it!